### PR TITLE
Change to V3 swagger and live API

### PR DESCRIPTION
**Pull request by RichardSmithONS**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -6,17 +6,17 @@ ENV FLASK_ENV=development
 ENV PORT=5000
 ENV HOST='0.0.0.0'
 # host.docker.internal wouldn't be required if we ran the API in another container
-ENV API_URL='http://host.docker.internal:9000'
-#ENV SWAGGER_URL=$API_URL'/assets/swagger.json'
-ENV SWAGGER_PATH='/home/aims/ai-swagger.json'
-ENV SECRET_KEY='you-will-never-guess'
+ENV API_URL='https://initial-test-bulk-1.aims.gcp.onsdigital.uk'
```
Comment (steve-thorne-ONS) [inline Dockerfile:9]: I don't think you should hardcode this. It can be injected at deploy time.

File: src/aims_dev_ui/config/dev.py [hunk 2]
```diff
@@ -4,8 +4,9 @@
 
 SECRET_KEY = 'secretkey'
 
-API_URL = os.getenv('API_URL') or 'http://localhost:9000'
+#API_URL = os.getenv('API_URL') or 'http://localhost:9000'
+API_URL = os.getenv('API_URL') or 'https://initial-test-bulk-1.aims.gcp.onsdigital.uk'
```
Comment (steve-thorne-ONS) [inline src/aims_dev_ui/config/dev.py:8]: And here....

File: src/aims_dev_ui/routes.py [hunk 3]
```diff
@@ -15,7 +15,7 @@
 
 def get_swagger_by_url():
   url = app.config['SWAGGER_URL']
-  response = requests.get(url)
+  response = requests.get(url, headers={'Authorization': 'Bearer ' + app.config['SECRET_KEY']})
```
Comment (steve-thorne-ONS) [inline src/aims_dev_ui/routes.py:18]: This will depend on the API being called.

File: src/aims_dev_ui/routes.py [hunk 4]
```diff
@@ -74,7 +74,7 @@ def endpoints(endpoint_path):
       uri = api_url + endpoint_value
 
     if form.bodyquery.data:
-      header = {"Content-Type": "application/json"}
+      header = {"Content-Type": "application/json", "Authorization": "Bearer " + app.config["SECRET_KEY"]}
```
Comment (steve-thorne-ONS) [inline src/aims_dev_ui/routes.py:77]: As above.

File: src/aims_dev_ui/routes.py [hunk 5]
```diff
@@ -100,7 +100,7 @@ def endpoints(endpoint_path):
         results = error
     else:
       try:
-        response = requests.get(uri, params=form.data)
+        response = requests.get(uri, params=form.data, headers={'Authorization': 'Bearer ' + app.config['SECRET_KEY']})
```
Comment (steve-thorne-ONS) [inline src/aims_dev_ui/routes.py:103]: and again....

Comment (RichardSmithONS) [reply src/aims_dev_ui/routes.py:103]: I have created a separate ticket for the exeter api which (wrongly) has Basic Auth. To apply this now would be jumping the gun.

File: src/aims_dev_ui/config/dev.py (hunk 2 above)
Comment (RichardSmithONS) [reply src/aims_dev_ui/config/dev.py:8]: I have created a separate ticket to parameterise the API URL as part of the concourse job. For now this is the only URL to use.

Comment (steve-thorne-ONS) [reply src/aims_dev_ui/config/dev.py:8]: The ENV variable will be replaced during deploy in the terraform.
I also don't think including the URL in the repo is good practice either.

File: src/aims_dev_ui/routes.py (hunk 5 above)
Comment (steve-thorne-ONS) [reply src/aims_dev_ui/routes.py:103]: This can be done in a new PR as it's not required yet. Nice and simple to implement.

**Review by steve-thorne-ONS — APPROVED** [review]
LGTM

**Pull request merged**